src/engine/GroupBy.cpp

```diff
@@ -383,23 +374,33 @@ ProtoResult GroupBy::computeResult(bool requestLaziness) {
   if (!subresult->isFullyMaterialized()) {
     AD_CORRECTNESS_CHECK(metadataForUnsequentialData.has_value());
 
-    auto localVocabPointer =
-        std::make_shared<LocalVocab>(std::move(localVocab));
-    cppcoro::generator<IdTable> generator = CALL_FIXED_SIZE(
+    cppcoro::generator<Result::IdTableVocabPair> generator = CALL_FIXED_SIZE(
         (std::array{inWidth, outWidth}), &GroupBy::computeResultLazily, this,
         std::move(subresult), std::move(aggregates),
         std::move(metadataForUnsequentialData).value().aggregateAliases_,
-        std::move(groupByCols), localVocabPointer, !requestLaziness);
+        std::move(groupByCols), !requestLaziness);
 
-    return requestLaziness
-               ? ProtoResult{std::move(generator), resultSortedOn(),
-                             std::move(localVocabPointer)}
-               : ProtoResult{cppcoro::getSingleElement(std::move(generator)),
-                             resultSortedOn(), std::move(*localVocabPointer)};
+    if (requestLaziness) {
+      return ProtoResult{std::move(generator), resultSortedOn()};
+    }
+    Result::IdTableVocabPair pair =
```

Comment:
What I am wondering,
Do we need both pairs (once with a shared ptr<localVocab> and once witha plain localVocab<...>)?
I mean, the semantics of `LocalVocab` are completely shared anyway, aren't they?
Then places like this can be much simpler. (Of course you also need the constructors for te result class that take the single pair.